Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -61,6 +61,13 @@ public class BufferedStream: Stream {
     /// Access this value only while `lock` is locked, to prevent simultaneous access.
     private var _buffer: Data
 
+    /// The number of bytes currently in the buffer, awaiting read.
+    public var bufferCount: Int {
```

Comment:
Added this `bufferCount` property (to see if the stream is holding any data in-memory) and a little refactoring in this type.